Sources/BoltsSwift/Task.swift

```diff
@@ -226,6 +226,46 @@ public final class Task<TResult> {
     }
 
     /**
+     Enqueues a given closure to be run once this task completes with error.
+
+     - parameter executor:     Determines how the the closure is called. The default is to call the closure immediately.
+     - parameter continuation: The closure that returns a task to chain on.
+
+     - returns: A task that will be completed when a task returned from a closure is completed.
+     */
+    public func continueOnErrorWith(executor: Executor = .Default, continuation: (ErrorType throws -> TResult)) -> Task {
```

Comment:
This should have a generic type, and not use TResult....

Also, can we have an overload that takes a specific error type defined by a generic parameter? It makes catching specific kinds of errors super easy.